command_handler.py

```diff
@@ -28,30 +28,31 @@ def validate_paths(dpct_project_path, destination_path):
     generates exceptions if directory of destination_path is not empty
     or if dpct_project_path directory contains no (nested) .dp.cpp or .dp.h files
     '''
-    if os.path.exists(dpct_project_path) is False:
-        r = "The dpct project path does not exist."
-        return r
-    filenames = os.listdir(dpct_project_path)
+    try:
+        if os.path.exists(dpct_project_path) is False:
+            raise Exception(dpct_project_path + " does not exist.")
+    except Exception as e:
+        return e
+
+    filenames = []
+    for root, dirs, files in os.walk(dpct_project_path):
+        filenames.extend(files)
     state_cpp = False
     for filename in filenames:
-        if filename.endswith('.cpp') is True:
+        if filename.endswith('.cpp') or filename.endswith('.hpp'):
             state_cpp = True
 
-    if not state_cpp:
-        r = "The path does not contain the cpp file."
-        return r
-    elif os.path.exists(destination_path) is False:
-        r = "This path does not exist."
-        return r
-    elif os.path.isdir(destination_path) is False:
-        r = "This is not a folder."
-        return r
-    elif os.listdir(destination_path):
-        r = "This folder is not empty."
-        return r
-    else:
-        print("Works!")
-        return True
+    try:
+        if not state_cpp:
+            raise Exception(dpct_project_path + " does not contain the cpp or hpp file.")
+        elif os.path.exists(destination_path) is False:
+            raise Exception(destination_path + " does not exist")
+        elif os.path.isdir(destination_path) is False:
+            raise Exception(destination_path + " is not a folder.")
+        elif os.listdir(destination_path):
+            raise Exception(destination_path + " is not empty.")
+    except Exception as e:
+        return e
```

Comment:
```suggestion
      if not state_cpp:
          raise Exception(dpct_project_path + " does not contain the cpp or hpp file.")
      elif os.path.exists(destination_path) is False:
          raise Exception(destination_path + " does not exist")
      elif os.path.isdir(destination_path) is False:
          raise Exception(destination_path + " is not a folder.")
      elif os.listdir(destination_path):
          raise Exception(destination_path + " is not empty.")
```
Again, no need for try except